Add damage/healing handler (for now)

placuszek15 opened the pull request.

placuszek15 committed:
Update README.md

placuszek15 committed:
Update main.py

shrianshChari commented on src/main.py:
```diff
@@ -148,9 +150,16 @@ def analyze_line(line: str) -> str:
     woke_up_pat = re.compile("(.*) woke up!")
 
+    damage_dealt_pat = re.compile("[0-9\.]*%")
+    super_effective_pat = re.compile("It's super effective!")
+    not_very_effective_pat = re.compile("It's not very effective...")
```
If we're just matching strings, without trying to capture anything, then we don't need to compile extra regex patterns.

shrianshChari commented on src/main.py:
```diff
@@ -281,6 +290,28 @@ def analyze_line(line: str) -> str:
+        move,use_player,target_player,use_mon,target_mon = moves_buffer
+        target_mon.heal(50.0)
+        converted = f"|-heal|p{target_player+1}a: {target_mon.nick}|{target_mon.approx_hp()}\/100"
+    elif not_very_effective_pat.match(line):
```
Make this an `==` instead of a `.match`

shrianshChari commented on src/main.py:
```diff
@@ -281,6 +290,28 @@ def analyze_line(line: str) -> str:
+        converted = f'|-resisted|p{target_player+1}a: {target_mon.nick}'
+
+
+    elif super_effective_pat.match(line):
```
Make this an `==` as well instead of a `.match`

shrianshChari commented on src/main.py:
```diff
@@ -354,7 +386,8 @@ def analyze_line(line: str) -> str:
                     full_msg[0] == players[1].name):
-                full_msg[0] = '☆' + full_msg[0]
+                #full_msg[0] = '☆' + full_msg[0]
+                full_msg[0] = '*' + full_msg[0]
```
I'm currently using the star character that Showdown already uses (indicating a player who is battling is speaking) instead of the asterisk (which indicates a bot on Showdown).

shrianshChari reviewed (changes requested):
Personally feel that you should revert the star for the chat messages and remove the extra regex objects.

placuszek15 committed:
Update parser

Change up so main.py is only a wrapper script around the gamesparser.py, added setup, immunity support and added the changes on the last pr

shrianshChari commented:
I'm not merging this "gamesparser.py" stuff, just do the mechanics first.

placuszek15 committed:
Revert "Update parser"

This reverts commit 7bf9a5703d5287790ef34f6b08b6d595bfe7f264.

placuszek15 committed:
Update main.py

shrianshChari commented:
Thank you!
